File: python/3254.find-the-power-of-k-size-subarrays-i.py
class Solution:
    def resultsArray(self, nums: list[int], k: int) -> list[int]:
        if k == 1:
            return nums

        prev = nums[0] - 1
        res = []
        counter = 1

        for i, cur in enumerate(nums):
            if prev + 1 == cur:
                counter += 1
            else:
                counter = 1

            prev = cur
            if i + 1 < k:
                continue

            if counter >= k:
                res.append(cur)
            else:
                res.append(-1)

        return res

File: cpp/3254.find-the-power-of-k-size-subarrays-i.cpp
#include <vector>
using std::vector;

class Solution {
public:
  vector<int> resultsArray(vector<int> &nums, int k) {
    if (k == 1) {
      return nums;
    }

    int counter = 1;
    int n = nums.size();
    vector<int> res;

    for (int i = 1; i < n; i++) {
      if (nums[i - 1] + 1 == nums[i]) {
        counter++;
      }

      if (i >= k) {
        if (nums[i - k] + 1 == nums[i - k + 1]) {
          counter--;
        }
      }

      if (i + 1 >= k) {
        if (counter == k) {
          res.push_back(nums[i]);
        } else {
          res.push_back(-1);
        }
      }
    }

    return res;
  }
};